model/DataObject.php

```diff
@@ -1717,7 +1717,8 @@ public function getRemoteJoinField($component, $type = 'has_many', &$polymorphic
 	 * @todo Implement query-params
 	 */
 	public function getManyManyComponents($componentName, $filter = "", $sort = "", $join = "", $limit = "") {
```

Comment:
You're right... but given you've done so much it seems like something pretty simple to fix.

Just change the params to `null` and if any of them `!== null` then throw a deprecation error.